src/app/static/src/app/store/modelCalibrate/mutations.ts

```diff
@@ -146,7 +146,15 @@ export const mutations: MutationTree<ModelCalibrateState> = {
 
     [ModelCalibrateMutation.ResetIds](state: ModelCalibrateState) {
         stopPolling(state)
-    }
+    },
+
+    [ModelCalibrateMutation.CalibrateResultFetched](state: ModelCalibrateState, action: PayloadWithType<CalibrateResultWithType>) {
+        if (!state.result) {
+            state.result = action.payload
+        } else {
+            state.result.data = [...state.result.data, ...action.payload.data]
```

Comment:
```suggestion
            state.result = structureClone(action.payload);
        } else {
            state.result.data.push(...action.payload.data);
```
So the `...` spread operator creates a shallow copy (copies over references to a new reference and doesnt create new memory addresses for all the content) but it is inefficient especially as a the size of `state.result.data` grows. Best to do a .push on it which wont create a copy.

The structure clone is there because something is setting the `action.payload` in the first if branch as inextensible (cant push to it basically) and this might explain your earlier struggles with just trying to push stuff, a structureClone will create a deep copy (new memory location for everything) and will also make it extensible, since we are only doing this deep copy once on load it should be completely fine!

Reply:
:pray: awesome thank you! I was fighting with .push